tests/unit/internal-session-test.js

```diff
@@ -189,6 +196,38 @@ describe('InternalSession', () => {
     describe('when the restored data does not contain an authenticator factory', () => {
       itDoesNotRestore();
     });
+
+    describe('with older synchronous stores (< v1.1.0)', function() {
+      describe('when restoring the session', function() {
+        describe('when the store resolves restoration', function() {
+          beforeEach(() => {
+            sinon.stub(store, 'restore').returns({ authenticated: { authenticator: 'authenticator' } });
+            sinon.stub(store, 'persist').returns();
+            sinon.stub(store, 'clear').returns();
+          });
+
+          it('is authenticated', () => {
+            return session.restore().then(() => {
+              expect(session.get('isAuthenticated')).to.be.true;
+            });
+          });
+        });
+
+        describe('when the store rejects restoration', function() {
```

Comment:
I added some tests to support this. Making sure session remains authenticated when `_clear` rejects is a bit hard, because `_clear` happens after `invalidate` and this cannot be switched as invalidation might need the data. An option could be introducing a `revalidate` method on authenticators in that case. Not sure if that's worth it. At this point invalidation rejects when `_clear` rejects but the session is still being invalidated.